iot_verifier/src/runner.rs

```diff
@@ -269,12 +264,37 @@ impl Runner {
                         }
                         return Ok(());
                     };
-
-                    let witness_reward_units = poc_challengee_reward_unit(
-                        verified_witnesses_result.verified_witnesses.len() as u32,
+                    // split our verified witness list up into selected and unselected items
+                    let max_witnesses_per_poc = self.settings.max_witnesses_per_poc as usize;
+                    let beacon_id = beacon.report_id(beacon_received_ts);
+                    let mut selected_witnesses = verified_witnesses_result
+                        .verified_witnesses
+                        .into_iter()
+                        .filter(|witness| witness.invalid_reason != InvalidReason::SelfWitness)
+                        .collect();
+                    let unselected_witnesses = shuffle_and_split_witnesses(
+                        &beacon_id,
+                        &mut selected_witnesses,
+                        max_witnesses_per_poc,
                     )?;
 
-                    verified_witnesses_result.update_reward_units(witness_reward_units);
+                    // get the number of valid witnesses in our selected list
+                    let num_valid_selected_witnesses = selected_witnesses
+                        .clone()
+                        .into_iter()
```

Comment:
should be able to do this without clone/into_iter; i would think counting the results wouldn't need to consume the iterator and counting on the references in the `iter()` version would work